tensorflow/core/ops/array_ops.cc

```diff
@@ -410,6 +410,20 @@ axis: Dimension along which to unpack.  Negative values wrap around, so the
 output: The list of tensors unpacked from `value`.
 )doc");
 
+REGISTER_OP("BroadcastTo")
+    .Input("input: T")
+    .Input("shape: int32")
+    .Output("output: T")
+    .Attr("T: type")
+    .SetShapeFn([](InferenceContext* c) { return Status::OK(); })
```

Comment:
You can do much more than a pass-through shape inference function here.

Specifically, if the shape tensor has a known constant value (and it often will) you should be able to set the output shape to be exactly that shape. See the shape_fn for reshape; you might be able to reuse most of its code here.

Reply:
Done. Thanks.